rest/route/host.go

```diff
@@ -312,3 +312,64 @@ func getLimit(vals url.Values) (int, error) {
 
 	return limit, nil
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// GET /host/range
+
+func makeFetchHostRange(sc data.Connector) gimlet.RouteHandler {
+	return &hostRangeGetHandler{
+		sc: sc,
+	}
+}
+
+type hostRangeGetHandler struct {
+	params model.APIHostParams
+	sc     data.Connector
+}
+
+func (h *hostRangeGetHandler) Factory() gimlet.RouteHandler {
+	return &hostRangeGetHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *hostRangeGetHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	if err := util.ReadJSONInto(body, &h.params); err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+
+	return nil
+}
+
+func (h *hostRangeGetHandler) Run(ctx context.Context) gimlet.Responder {
+	dbUser := MustHaveUser(ctx)
+	username := ""
+	// only admins see hosts that aren't theirs
+	if !util.StringSliceContains(h.sc.GetSuperUsers(), dbUser.Username()) || h.params.Mine {
+		username = dbUser.Username()
+	}
+
+	hosts, err := h.sc.FindHostsInRange(h.params, username)
+	if err != nil {
+		gimlet.NewJSONErrorResponse(errors.Wrap(err, "Database error"))
+	}
+
+	resp := gimlet.NewResponseBuilder()
+	if err = resp.SetFormat(gimlet.JSON); err != nil {
+		return gimlet.MakeJSONErrorResponder(err)
+	}
+	for _, host := range hosts {
```

Comment:
does this end up getting paginated?

Reply:
not as implemented.
I assumed: 1) the list of hosts would be able to be transferred in the time before we timeout and 2) it wasn't worthwhile to break up the list on the client side and have the user ask for more.
I would be open to adding pagination...